### PR TITLE
KTO changes to return aux outputs

### PR DESCRIPTION
**Pull request by vaibhavjindal**
## Summary

This PR introduces the following changes to enable integration with huggingface TRL:
1. KTO loss can now return the following along with the loss:
- chosen_logps
- rejected_logps
- sum(chosen_logits)
- sum(rejected_logits)
- chosen_rewards
- rejected_rewards
2. Adds an option to enable/disable log_probs averaging while calculating the loss.

## Details
Benchmark results with the new implementation:
kto_loss_memory
kto_loss_speed

## Testing Done

- Hardware Type: <BLANK>
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [ ] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by hebiao064** [comment]
Can you run a benchmark to see if the memory/speed still performs at certain level?

**Comment by vaibhavjindal** [comment]
> Can you run a benchmark to see if the memory/speed still performs at certain level?

Sure, will add it.

File: src/liger_kernel/chunked_loss/fused_linear_unpaired_preference.py [hunk 1]
```diff
@@ -218,29 +293,60 @@ def _compute_loss(
...
+        preference_loss_outputs = preference_loss_fn(
+            log_prob_chunk, preference_labels_chunk, full_target, **loss_kwargs
+        )
+        if isinstance(preference_loss_outputs, tuple):
+            preference_loss_chunk, *aux_outputs = preference_loss_outputs
+        else:
+            preference_loss_chunk, aux_outputs = preference_loss_outputs, []
+
+        chosen_logits_sum = chosen_logits.nansum()
+        rejected_logits_sum = rejected_logits.nansum()
```
Comment (shivam15s) [inline src/liger_kernel/chunked_loss/fused_linear_unpaired_preference.py:343]: do you expect these to have nans?

Comment (vaibhavjindal) [reply src/liger_kernel/chunked_loss/fused_linear_unpaired_preference.py:343]: I was a bit unsure about this as TRL is using nansum() for this: https://github.com/huggingface/trl/blob/491921c1a4167e7c84429382470b0bb3158e66b0/trl/trainer/kto_trainer.py#L1271. Thus kept the nansum() to deal with the worst case.

**Review by shivam15s — APPROVED** [review]
lgtm, lets figure out the speed drop in the next PR

**Review by hebiao064 — APPROVED** [review]
